src/printer/printer.h

```diff
@@ -137,7 +137,7 @@ class Printer
                                    const std::vector<Node>& vars,
                                    TypeNode range,
                                    bool isInv,
-                                   TypeNode sygusType) const;
+                                   TypeNode* sygusType = nullptr) const;
```

Comment:
I would prefer not using TypeNode*, which is conceptually a double pointer.  You can do `TypeNode sygusType = TypeNode::null()` here.